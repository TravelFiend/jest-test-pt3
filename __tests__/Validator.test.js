const Validator = require('../lib/Validator');

describe('Validator', () => {
    let spokeValidator;
    
    describe('required fields', () => {
        beforeAll(() => {
            spokeValidator = new Validator('spokes', {
                type: Number,
                required: true
            });
        });

        it('should return value if field is present and correctly typed', () => {
            const bike = {
                brand: 'Yeti',
                spokes: 36,
                saddle: 'leather'
            };
            console.log(spokeValidator.validate(bike));
            
            expect(spokeValidator.validate(bike)).toEqual(36);
        });

        it('should return value if field is present and incorrectly typed', () => {
            const bike = {
                brand: 'Yeti',
                spokes: '36',
                saddle: 'leather'
            };

            expect(spokeValidator.validate(bike)).toEqual(36);
        });

        it('should throw if cannot cast, what were thinking you dimb', () => {
            const bike = {
                brand: 'Yeti',
                spokes: 'dimby',
                saddle: 'leather'
            };

            expect(() => spokeValidator.validate(bike)).toThrowErrorMatchingSnapshot();
        });

        it('should throw if field is missing, be better at things', () => {
            const bike = {
                brand: 'Yeti',
                saddle: 'leather'
            };

            expect(() => spokeValidator.validate(bike)).toThrowErrorMatchingSnapshot();
        });
    });

    describe('not required fields', () => {
        beforeAll(() => {
            spokeValidator = new Validator('spokes', {
                type: Number,
            });
        });

        it('should return value if field is present and correctly typed', () => {
            const bike = {
                brand: 'Yeti',
                spokes: 36,
                saddle: 'leather'
            };

            expect(spokeValidator.validate(bike)).toEqual(36);
        });

        it('should return value if field is present and incorrectly typed', () => {
            const bike = {
                brand: 'Yeti',
                spokes: '36',
                saddle: 'leather'
            };

            expect(spokeValidator.validate(bike)).toEqual(36);
        });

        it('should throw if cannot cast, what were thinking you dimb', () => {
            const bike = {
                brand: 'Yeti',
                spokes: 'dimby',
                saddle: 'leather'
            };

            expect(() => spokeValidator.validate(bike)).toThrowErrorMatchingSnapshot();
        });

        it('should throw if field is missing, be better at things', () => {
            const bike = {
                brand: 'Yeti',
                saddle: 'leather'
            };

            expect(spokeValidator.validate(bike)).toBeNull();
        });
    });
});
