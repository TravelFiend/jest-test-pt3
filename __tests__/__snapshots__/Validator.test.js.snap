// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator not required fields should throw if cannot cast, what were thinking you dimb 1`] = `"Cannot cast >>dimby<< to Number"`;

exports[`Validator required fields should throw if cannot cast, what were thinking you dimb 1`] = `"Cannot cast >>dimby<< to Number"`;

exports[`Validator required fields should throw if field is missing, be better at things 1`] = `"You're missing the spokes field"`;
